systemd/crc-needs-tap.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -o pipefail
+set -o errexit
+set -o nounset
+set -o errtrace
+set -x
+
+source /etc/sysconfig/crc-env || echo "WARNING: crc-env not found"
+
+running_on_aws() {
+    # Set a timeout for the curl command
+    TIMEOUT=1
+
+    # Check the DMI registry
+    if dmidecode | grep -q "Amazon EC2"; then
+        echo "✅ Running on an AWS EC2 instance (detected via DMI)."
+        return 0
+
+    # As a fallback, check the metadata service
+    elif curl -s -m $TIMEOUT http://169.254.169.254/latest/meta-data/instance-id > /dev/null 2>&1; then
+        echo "✅ Running on an AWS EC2 instance."
+        return 0
+    else
+        echo "❌ Not running on an AWS EC2 instance."
+        return 1
+    fi
+}
```

Comment:
I think we have discussed that tap device shouldn't depend on the cloud provider because user can use cloud provider to provision bare metal machine and then use crc for cluster creation. 

- https://github.com/crc-org/crc/issues/4909